beacon_node/execution_layer/src/lib.rs

```diff
@@ -454,12 +455,28 @@ impl<T: EthSpec> ExecutionLayer<T> {
     }
 
     /// Cache a full payload, keyed on the `tree_hash_root` of the payload
-    fn cache_payload(&self, payload: ExecutionPayloadRef<T>) -> Option<ExecutionPayload<T>> {
-        self.inner.payload_cache.put(payload.clone_from_ref())
+    fn cache_payload(
+        &self,
+        payload_and_blobs: PayloadContentsRefTuple<T>,
+    ) -> Option<FullPayloadContents<T>> {
+        let (payload_ref, maybe_json_blobs_bundle) = payload_and_blobs;
+
+        let payload = payload_ref.clone_from_ref();
+        let maybe_blobs_bundle = maybe_json_blobs_bundle
+            .cloned()
+            .map(|blobs_bundle| BlobsBundle {
+                commitments: blobs_bundle.commitments,
+                proofs: blobs_bundle.proofs,
+                blobs: blobs_bundle.blobs,
+            });
+
+        self.inner
+            .payload_cache
```

Comment:
`PayloadCache` is an LruCache with default capacity = 10. Need to consider if this is still a good default now we're caching and intentionally keeping a copy of local EL blobs. We no longer `pop` the payload from this cache since #4239. 

https://github.com/sigp/lighthouse/blob/744b1950e54a4cf80c91e1f3a8dc26128deab062/beacon_node/execution_layer/src/payload_cache.rs#L6

Reply:
I think this is still alright, the max cache size would still only be something like 10MB if my math is right

Reply:
Yep that sounds about right! Thanks